clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,8 +497,22 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
-
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
+        new ApiExceptionBuilder() {
+            @Override
+            public ApiException build(String message) {
+                return new KafkaStorageException(message);
+            }
+    }),
+    UNKNOWN_ERROR_CODE(57, "The client received an unknown error code when processing the response. This may happen if " +
+            "the client library version is lower than the server library version",
+        new ApiExceptionBuilder() {
```

Comment:
Hmm, if a new error is not retriable, the server code will probably look like the following.

```
// Check if the client recognizes NEW_NON_RETRIABLE_ERROR_CODE
if (version < 5 and original_error_code = NEW_NON_RETRIABLE_ERROR_CODE) // 5 is the version that understands NEW_NON_RETRIABLE_ERROR_CODE
  return unknown_server_error_code
else
  return NEW_NON_RETRIABLE_ERROR_CODE
```
So, you don't really save code if the new error is not retriable. Another thing that bothers me a bit is that the logic for handling a retriable error is a bit different from that for handing a non-retriable error. In the alternative approach, at least any new error code is handled in the same way.